tests/test_proc_macro_hygiene.rs

```diff
@@ -0,0 +1,66 @@
+#![no_implicit_prelude]
+
+macro_rules! shadow {
+    ($name: ident) => {
+        ::paste::item! {
+            #[allow(non_camel_case_types, dead_code)]
+            unsafe trait [<NobodyImplsThis_ $name>]  {}
+
+            #[allow(non_camel_case_types, dead_code)]
+            struct [<Shadows_ $name>]<T: [<NobodyImplsThis_ $name>]> {
+                _ty: ::core::marker::PhantomData<T>,
+              }
+
+            #[allow(non_camel_case_types, dead_code)]
+            type $name = [<Shadows_ $name>]<()>;
+        }
+    };
+}
+
+shadow!(std);
+shadow!(alloc);
+shadow!(core);
+shadow!(pyo3);
```

Comment:
All this shadowing is unneeded - with the `no_implicit_prelude` all the crates aren't available anyway and you'll get `use of undeclared crate or module`.

Reply:
(if that is indeed the case, that would mean we also wouldn't be dependent on the `paste!` macro in this test, which would be nice)

Reply:
Thanks, I didn't realize that. I've removed it. 👍